### PR TITLE
Bjorncs/custom ssl context

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

**Comment by bjorncs** [comment]
Ignore the branch name. It's not related to the content of this PR. This PR is another prerequisite for custom ssl context configuration, hence the original branch name :)

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/server/jetty/ConnectorFactory.java [hunk 1]
```diff
@@ -54,12 +52,30 @@ private static void validateSslConfig(ConnectorConfig config) {
...
         if (ssl.keyStoreType() == PEM) {
-            if (! ssl.keyStorePath().isEmpty())
+            if (! ssl.keyStorePath().isEmpty()) {
                 throw new IllegalArgumentException("keyStorePath can not be set when keyStoreType is PEM");
+            }
+            if (!ssl.keyDbKey().isEmpty()) {
+                // TODO Make an error once there are separate passwords for truststore and keystore
+                log.warning("Encrypted PEM key stores are not supported. Password is only applied to truststore");
+            }
+            if (ssl.pemKeyStore().certificatePath().isEmpty()) {
```
Comment (gjoranv) [inline jdisc_http_service/src/main/java/com/yahoo/jdisc/http/server/jetty/ConnectorFactory.java]: Braces, braces and more braces...

**Review by gjoranv — APPROVED** [review]

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/server/jetty/ConnectorFactory.java [hunk 2]
```diff
@@ -54,12 +52,30 @@ private static void validateSslConfig(ConnectorConfig config) {
         ConnectorConfig.Ssl ssl = config.ssl();
 
         if (ssl.keyStoreType() == JKS) {
-            if (! ssl.pemKeyStore().keyPath().isEmpty() || ! ssl.pemKeyStore().certificatePath().isEmpty())
+            if (!ssl.pemKeyStore().keyPath().isEmpty() || ! ssl.pemKeyStore().certificatePath().isEmpty()) {
```
Comment (gjoranv) [inline jdisc_http_service/src/main/java/com/yahoo/jdisc/http/server/jetty/ConnectorFactory.java]: Inconsistent use of whitespace around '!'

**Comment by gjoranv** [comment]
👍

**Pull request merged**